libr/bin/p/bin_dex.c

```diff
@@ -762,6 +801,29 @@ static void parse_class(RBinFile *binfile, RBinDexObj *bin, RBinDexClass *c, int
 				try_item[tries_size] tries
 				encoded_catch_handler_list handlers
 #endif
+
+
+				// TODO: parse debug info
+				// TODO: check for bof
+				ut16 regsz = r_read_le16(binfile->buf->buf+MC);
```

Comment:
this line and the following ones can produce an OOB read. you should not access the binfile->buf->buf directly because binfile->buf is an RBuffer instance, that requires access via an API, that api control buffer boundaries and allows to read safe blocks from there. Also that buffer can contain mmap, malloc, fd or raw buffer pointer.. so use r_buf_get_at() or r_buf_read() for safe reads